sql/src/main/java/org/apache/druid/sql/http/SqlQuery.java

```diff
@@ -36,7 +36,10 @@
   public static List<TypedValue> getParameterList(List<SqlParameter> parameters)
   {
     return parameters.stream()
-                     .map(SqlParameter::getTypedValue)
+                     // null params are not good!
+                     // we pass them to the planner, so that it can generate a proper error message.
+                     // see SqlParameterizerShuttle and RelParameterizerShuttle.
+                     .map(p -> p == null ? null : p.getTypedValue())
```

Comment:
Can we throw exception from here?

Reply:
The intention was passing null parameters as they are so that `SqlParameterizerShuttle` and `RelParameterizerShuttle` can handle them properly. They will throw `InvalidArgumentException` at https://github.com/apache/druid/pull/11900/files#diff-ae4b6e9293a80177b076ab1a7d6235931eb95619dcc759ae40027543ef35f4e1R215 or https://github.com/apache/druid/pull/11900/files#diff-9ea07312c8d034ebe3c983a4690dd3cb79a2aa190610b11583cf37947f919de9R75. Do you think it's better to throw an exception here too? I thought it would be better to put error handling logic in one place.

Reply:
Actually, parameters cannot be null in production code path when you use SQL over HTTP. This bug exists only for JDBC which doesn't use this method. This change was for easily mimicking the error case in unit tests.